### DISCUSSION TITLE
Can fltk-rs can easily be binded to a game engine?

### DISCUSSION THREAD

**Discussion by Raj2032** [fltk-rs/fltk-rs#1193] [Q&A] [answered]
Hi, just like how egui can be easily integrated into a game engine, is this the same with fltk-rs by any chance?

**Comment by MoAlyousef** [comment, marked as answer]
Not really since FLTK does it’s own windowing. It’s technically possible, and there are guides on integrating FLTK (C++) into existing engines like Ogre3D, but it involves calling platform specific functions to reparent the FLTK window, there’s also an experimental GlWidgetWindow which allows drawing FLTK widgets on an OpenGL context (using OpenGL drawing routines).

**Reply by Raj2032** [reply]
> and there are guides on integrating FLTK (C++)

Would this work with fltk-rs as well?

**Reply by MoAlyousef** [reply]
It should but I haven’t tried it with any of the Rust engines.

**Reply by Raj2032** [reply]
I see thanks
